scipy/optimize/tests/test_direct.py

```diff
@@ -0,0 +1,40 @@
+"""
+Unit test for DIRECT optimization algorithm.
+"""
+from numpy.testing import (assert_, assert_array_almost_equal,
+                           assert_allclose, assert_equal)
+from pytest import raises as assert_raises
+import pytest
+import numpy as np
+
+from scipy.optimize import minimize
+
+class TestDIRECT:
+
+    def get_test_func(self, name):
```

Comment:
- [ ] test for `nfev` is correct
- [ ] test that res.success is False if maxfun exceeded.
- [ ] test that res.success is False if maxiter reached.
- [ ] test what happens if an exception is raised in the user function (it should be raised to the user code, not segfault).
- [ ] test what happens if function returns np.inf, etc. 
- [ ] add an assertion to the test function that `x` *strictly* lies in between `lb, ub`.

Reply:
Addressed.